spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -302,6 +336,74 @@ def sample_types_from(decoded_profile)
       end
     end
 
+    describe 'heap samples' do
+      let(:metric_values1) { { 'cpu-time' => 101, 'cpu-samples' => 1, 'wall-time' => 789, 'alloc-samples' => 10 } }
+      let(:metric_values2) { { 'cpu-time' => 102, 'cpu-samples' => 2, 'wall-time' => 790, 'alloc-samples' => 11 } }
+      let(:labels) { { 'label_a' => 'value_a', 'label_b' => 'value_b', 'state' => 'unknown' }.to_a }
+      let(:sample_rate) { 50 }
+
+      let(:a_string) { 'a beautiful string' }
+      let(:another_string) { 'a fearsome string' }
+      let(:an_array) { [1..10] }
+      let(:another_array) { [-10..-1] }
```

Comment:
Btw I'm not sure if you really want this or `(1..10).to_a`. `1..10` is a literal for a range object so these arrays only have a single object:

```ruby
[5] pry(main)> [1..10].size
=> 1
[6] pry(main)> [1..10][0].class
=> Range
```

Reply:
Indeed! If it weren't for your eagle eyes, I would have only noticed it later when I added the size tests and maybe waste an hour trying to understand what was happening 😂